americangut/util.py

```diff
@@ -635,3 +636,58 @@ def get_single_id_lists(map_, depths):
             single_ids[depths[idx - 1]].extend(single_ids[depths[idx]])
 
     return single_ids
+
+
+def collapse_taxonomy(_bt, level=5):
+    """Collapses OTUs by taxonomy
+
+    Parameters
+    ----------
+    _bt : biom table
+        Table to collapse
+    level : int, optional
+        Level to collapse to. 0=kingdom, 1=phylum,...,5=genus, 6=species
+        Default 5
+
+    Returns
+    -------
+    biom table
+        Collapsed biom table
+
+    Citations
+    ---------
+    [1] http://biom-format.org/documentation/table_objects.html
+    """
+    def collapse_f(id_, md):
+        return '; '.join(md['taxonomy'][:level + 1])
+    collapsed = _bt.collapse(collapse_f, axis='observation', norm=False)
+    return collapsed
+
+
+def collapse_full(_bt):
```

Comment:
I tried that, but they need to collapse by sample not by taxa, and I don't think that's possible.